lightning/src/routing/network_graph.rs

```diff
@@ -0,0 +1,1672 @@
+		network.write(writer)?;
+		Ok(())
+	}
+}
+
+/// Arguments for the creation of a NetGraphMsgHandler that are not deserialized.
+/// At a high-level, the process for deserializing a NetGraphMsgHandler and resuming normal operation is:
+/// 1) Deserialize the NetGraphMsgHandler by filling in this struct and calling <NetGraphMsgHandler>::read(reaser, args).
+/// 2) Register the new NetGraphMsgHandler with your ChainWatchInterface
+pub struct NetGraphMsgHandlerReadArgs {
+	/// The ChainWatchInterface for use in the NetGraphMsgHandler in the future.
+	///
+	/// No calls to the ChainWatchInterface will be made during deserialization.
+	pub chain_monitor: Arc<ChainWatchInterface>,
+	/// The Logger for use in the ChannelManager and which may be used to log information during
+	/// deserialization.
+	pub logger: Arc<Logger>,
+}
+
+impl ReadableArgs<NetGraphMsgHandlerReadArgs> for NetGraphMsgHandler {
+	fn read<R: ::std::io::Read>(reader: &mut R, args: NetGraphMsgHandlerReadArgs) -> Result<NetGraphMsgHandler, DecodeError> {
+		let _ver: u8 = Readable::read(reader)?;
+		let min_ver: u8 = Readable::read(reader)?;
+		if min_ver > SERIALIZATION_VERSION {
+			return Err(DecodeError::UnknownVersion);
+		}
+		let network_graph = Readable::read(reader)?;
+		Ok(NetGraphMsgHandler {
+			secp_ctx: Secp256k1::verification_only(),
+			network_graph: RwLock::new(network_graph),
+			chain_monitor: args.chain_monitor,
+			full_syncs_requested: AtomicUsize::new(0),
+			logger: args.logger.clone(),
+		})
+	}
+}
+
+#[derive(PartialEq, Debug)]
+/// Details regarding one direction of a channel
+pub struct DirectionalChannelInfo {
+	/// When the last update to the channel direction was issued
+	pub last_update: u32,
+	/// Whether the channel can be currently used for payments
+	pub enabled: bool,
+	/// The difference in CLTV values between the source and the destination node of the channel
+	pub cltv_expiry_delta: u16,
+	/// The minimum value, which must be relayed to the next hop via the channel
+	pub htlc_minimum_msat: u64,
+	/// Fees charged when the channel is used for routing
+	pub fees: RoutingFees,
+	/// Most recent update for the channel received from the network
+	pub last_update_message: Option<msgs::ChannelUpdate>,
+}
+
+impl std::fmt::Display for DirectionalChannelInfo {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		write!(f, "last_update {}, enabled {}, cltv_expiry_delta {}, htlc_minimum_msat {}, fees {:?}", self.last_update, self.enabled, self.cltv_expiry_delta, self.htlc_minimum_msat, self.fees)?;
+		Ok(())
+	}
+}
+
+impl_writeable!(DirectionalChannelInfo, 0, {
+	last_update,
+	enabled,
+	cltv_expiry_delta,
+	htlc_minimum_msat,
+	fees,
+	last_update_message
+});
+
+#[derive(PartialEq)]
+/// Details regarding a channel (both directions)
+pub struct ChannelInfo {
+	/// Protocol features of a channel communicated during its announcement
+	pub features: ChannelFeatures,
+	/// Source node of one of the directions of a channel
```

Comment:
Maybe rename the variable to just be node_one and then its clear which direction goes with which node?